src/Build/BuildCheck/API/BuildExecutionCheck.cs

```diff
@@ -9,29 +9,29 @@
 namespace Microsoft.Build.Experimental.BuildCheck;
 
 /// <summary>
-/// Base class for build analyzers.
-/// Same base will be used for custom and built-in analyzers.
-/// <see cref="BuildAnalyzer"/> is a unit of build analysis execution, but it can contain multiple rules - each representing a distinct violation.
+/// Base class for build checks.
+/// Same base will be used for custom and built-in checks.
+/// <see cref="BuildExecutionCheck"/> is a unit of build checks execution, but it can contain multiple rules - each representing a distinct violation.
 /// </summary>
-public abstract class BuildAnalyzer : IDisposable
+public abstract class BuildExecutionCheck : IDisposable
```

Comment:
Lets keep BuildCheck.

To prevent conflicts with namespace - we can rename the namespace - e.g. to `Microsoft.Build.Experimental.BuildChecking`

Reply:
I agree to keep it Lets keep BuildCheck

Reply:
As discussed on the meeting yesterday, we are changing it to just `Check` to not confuse with our infrastructure.